### PR TITLE
Exception handling PoC

### PR DESCRIPTION
**Pull request by hadrienk**
This PR is in preparation for #40. The goal is to improve the exception handling in the java-script project.

I created a new `ContextualRuntimeException` that contains a reference to the ParserRuleContext we get from ANTLR so that we can give the position of an error to the user. I also added `VTLScriptException` which extends the ScriptException, mainly because the ScriptException constructors are too restricting.

I adjusted `AssignmentVisitor`, `ExpressionVisitor` and `NativeFunctionsVisitor` to use the new exceptions. Take a look at `VTLErrorsTest` for an example of what the result is.

### CODE REVIEW

File: java-vtl-script/src/main/java/no/ssb/vtl/script/VTLScriptEngine.java [hunk 1]
```diff
@@ -134,89 +115,57 @@ public Object eval(String script, ScriptContext context) throws ScriptException
...
-                        );
-                    }
-                    wrappedException.setLine(line);
-                    wrappedException.setColumn(column);
-                    throw new ParseCancellationException(msg, wrappedException);
-                }
-            };
-            lexer.addErrorListener(errorListener);
-            parser.addErrorListener(errorListener);
+    public VTLParser.StartContext compile(Reader reader, Consumer<VTLScriptException> errorConsumer) throws IOException {
```
Comment (pawbu) [inline java-vtl-script/src/main/java/no/ssb/vtl/script/VTLScriptEngine.java]: I'm not sure about the `compile` word here. In antlr it is used in the context of compiling a grammar file.

File: java-vtl-script/src/main/java/no/ssb/vtl/script/visitors/functions/NativeFunctionsVisitor.java [hunk 2]
```diff
@@ -138,6 +116,49 @@ public Class getVTLType() {
...
+        VTLExpression finalNullable = coerceNullLiteralType(nullable, replacement);
+
+        if (!finalNullable.getVTLType().equals(replacement.getVTLType())) {
+            throw new ContextualRuntimeException(
+                    format("%s and %s must be of the same type",
+                            nullableCtx.getText(), replacementCtx.getText()
+                    ), ctx
+            );
+        }
+//            checkArgument(
```
Comment (pawbu) [inline java-vtl-script/src/main/java/no/ssb/vtl/script/visitors/functions/NativeFunctionsVisitor.java]: Remove if not needed?

File: java-vtl-script/src/main/java/no/ssb/vtl/script/VTLScriptEngine.java (hunk 1 above)
Comment (hadrienk) [reply java-vtl-script/src/main/java/no/ssb/vtl/script/VTLScriptEngine.java]: right, how about validate?

Comment (pawbu) [reply java-vtl-script/src/main/java/no/ssb/vtl/script/VTLScriptEngine.java]: maybe `getRootContext` or `getStartContext`. Or maybe split this method as it sets up lexer and parser and also gets the start context.

**Pull request merged**